shell/src/main/java/org/apache/accumulo/shell/format/DeleterFormatter.java

```diff
@@ -79,33 +79,36 @@ public String next() {
     Mutation m = new Mutation(key.getRow());
     String entryStr = formatEntry(next, isDoTimestamps());
     boolean delete = force;
-    try {
-      if (!force) {
-        shellState.getReader().flush();
-        String line = shellState.getReader().readLine("Delete { " + entryStr + " } ? ");
-        more = line != null;
-        delete = line != null && (line.equalsIgnoreCase("y") || line.equalsIgnoreCase("yes"));
+    String line;
+    if (!force) {
+      try {
+        shellState.getWriter().flush();
+        String prompt = "Delete { " + entryStr + " } ? ";
+        line = shellState.getReader().readLine(prompt);
+      } catch (EndOfFileException ignored) {
+        // Reached the end of file. Line is null to keep old functionality.
+        line = null;
       }
-      if (delete) {
-        m.putDelete(key.getColumnFamily(), key.getColumnQualifier(),
-            new ColumnVisibility(key.getColumnVisibility()), key.getTimestamp());
-        try {
-          writer.addMutation(m);
-        } catch (MutationsRejectedException e) {
-          log.error(e.toString());
-          if (Shell.log.isTraceEnabled()) {
-            for (ConstraintViolationSummary cvs : e.getConstraintViolationSummaries()) {
-              log.trace(cvs.toString());
-            }
+      more = line != null;
+      delete = line != null && (line.equalsIgnoreCase("y") || line.equalsIgnoreCase("yes"));
```

Comment:
I think this suggestion was made previously, but may have been overlooked.
```suggestion
      delete = more && (line.equalsIgnoreCase("y") || line.equalsIgnoreCase("yes"));
```

Reply:
You are correct. I will add that in.